.pre-commit-config.yaml

```diff
@@ -0,0 +1,34 @@
+# exclude library folder from checks, except prometheus library itself
+exclude: ^((?!lib/charms/prometheus_k8s)lib).*$
+repos:
+  - repo: https://github.com/PyCQA/autoflake
+    rev: v1.7.6
+    hooks:
+      - id: autoflake
+  - repo: https://github.com/psf/black
+    rev: 22.3.0
+    hooks:
+      - id: black
+  - repo: https://github.com/asottile/blacken-docs
+    rev: v1.12.1
+    hooks:
+      - id: blacken-docs
+        additional_dependencies: [black]
+  - repo: https://github.com/pycqa/isort
+    rev: 5.10.1
+    hooks:
+      - id: isort
+        name: isort (python)
+
+  # official flake8 doesn't support pyproject.toml configuration.
+  - repo: https://github.com/csachs/pyproject-flake8
+    rev: v5.0.4
+    hooks:
+      - id: pyproject-flake8
+        name: pyproject-flake8
+
+  - repo: https://github.com/Lucas-C/pre-commit-hooks-nodejs
```

Comment:
> Does not require to have NodeJS installed: pre-commit will fetch & install it under the hood.

:scream: 

But also, we don't use `<!-- toc -->` comments yet. Probably better to leave this one off for now.

Reply:
JFYI

- hooks are provided by the software provider
- hooks do not require CLI args, you can, if you want, but as current PR shows, tools will rely on `pyproject.toml` if no argument is supplied


Overall, PR is backwards compatible. For those who do not want to use it, just use `tox -e fmt` and it will do the same thing. For those who want to autofix style in each commit, install via `pre-commit install`

PS
I just sent you a list of projects that use `tox` + `pre-commit` to show use case, but have a look for amount of projects that use only `pre-commit`. Almost every huge python project